Makefile

```diff
@@ -6,15 +6,32 @@ CROSS ?= linux/arm linux/arm64 linux/ppc64le linux/s390x \
 SUDO ?= sudo -n
 
 .PHONY: all
-all: lint test cross
+all: clean lint test test-local cross
+
+.PHONY: clean
+clean:
+	$(RM) mount/go-local.*
 
 .PHONY: test
 test: RUN_VIA_SUDO = $(shell $(SUDO) true && echo -exec \"$(SUDO)\")
-test:
+test: clean
 	for p in $(PACKAGES); do \
 		(cd $$p && go test $(RUN_VIA_SUDO) -v .); \
 	done
 
+# test the mount module against the local mountinfo source code instead of the
+# release specified in go.mod. This allows catching regressions / breaking
+# changes in mountinfo.
+.PHONY: test-local
+test-local: RUN_VIA_SUDO = $(shell $(SUDO) true && echo -exec \"$(SUDO)\")
+test-local: clean
+	echo 'replace github.com/moby/sys/mountinfo => ../mountinfo' | cat mount/go.mod - > mount/go-local.mod \
+	&& cd mount \
+	&& go mod download -modfile=go-local.mod \
+	&& go test -modfile=go-local.mod $(RUN_VIA_SUDO) -v . ;\
```

Comment:
nit: add a space before `\`.

Reply:
With your other comments about continuation; I'll remove this, and make the rm just `rm -f mount/go-local.*`